### DISCUSSION TITLE
Github Personal Tokens Expire after some time - give 403 error

### DISCUSSION THREAD

**Discussion by gita-vahdatinia** [community/community#24319] [Apps, API and Webhooks]
<p>I am using my github personal access token for a CI pipeline for entire organization. A couple issues:</p>
<ul>
<li>Is it possible to have a token used for the entire org rather than I use my personal token</li>
<li>The token that I am using expires after some time – I receive a 403 Client Error: Forbidden for url; I rotate the tokens and then it works fine. What can I do to prevent that?</li>
</ul>

**Comment by byrneh** [comment]
<p>There is no expiry time for a personal access token on an account that I am aware (other products like azure devops have this option). Its a feature thats been asked for by a number of people.</p>
<p>So seems strange you get a forbidden, the root cause for 403 forbidden probably needs investigation.</p>
<p>There is not a token you can attach to the Organization as such.</p>
<p>Where a token is needed for a machine/non-user account puspose like pipeline or batch script it’s common to setup another machine user account rather than use a personal user account. This mitigates any risk of another user obtaining yiur user credentials and acting as you on GitHub.</p>

**Comment by gita-vahdatinia** [comment]
<p>Ahh makes sense to create another user for toke purposes.<br>
Thanks</p>
<p>Yes, the 403 seemingly randomly occurs.</p>

**Comment by gita-vahdatinia** [comment, marked as answer]
<p>There is a 5k limit per hour for tokens – we kept hitting that limit.</p>

**Comment by rayan1360** [comment]
<p>37.129.180.106/+3:30/2612c6fe7ca46848</p>
